server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/FileAttributes.java

```diff
@@ -31,10 +33,13 @@
   private final boolean unitTest;
   @CheckForNull
   private final String languageKey;
+  private final int lines;
 
-  public FileAttributes(boolean unitTest, @Nullable String languageKey) {
+  public FileAttributes(boolean unitTest, @Nullable String languageKey, int lines) {
     this.unitTest = unitTest;
     this.languageKey = languageKey;
+    checkArgument(lines > 0, "Lines has not been set for this file");
```

Comment:
I can update the message, but I don't understand why should I put the check in ComponentRootBuilder